src/main/kotlin/com/zepben/evolve/metrics/NetworkContainerMetrics.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.metrics
+
+/**
+ * A map from metric names to their values.
+ */
+class NetworkContainerMetrics(private val map: MutableMap<String, Double> = mutableMapOf()) : MutableMap<String, Double> by map {
+
+    /**
+     * Increment the value of the metric named [key]. If no such metric exists, it is automatically created with an initial value of zero.
+     *
+     * @param key The metric to increment
+     * @return This object, for fluent use
+     */
+    fun inc(key: String): NetworkContainerMetrics = plus(key, 1.0)
+
+    /**
+     * Increase the value of the metric named [key] by [amount]. If no such metric exists, it is automatically created with an initial value of zero.
+     *
+     * @param key The metric to increase
+     * @param amount The amount to increase the metric by. A negative value will decrease the metric.
+     * @return This object, for fluent use
+     */
+    fun plus(key: String, amount: Number = 1.0): NetworkContainerMetrics = apply { merge(key, amount.toDouble(), Double::plus) }
```

Comment:
todo: Shouldn't have a default for the `amount` in `plus`, this is what `inc` is for. I had a copy/paste error in my previous comment from your original `increase` where the default made sense.

Reply:
Removed default and edited tests and changelog to match.